### PR TITLE
Replace bitcoind_api string formatting with URL crate to avoid missing/double '/' path errors

### PR DESCRIPTION
**Pull request by jferrant**
Closes https://github.com/Trust-Machines/core-eng/issues/427
Closes https://github.com/Trust-Machines/core-eng/issues/423

### CODE REVIEW

**Review by sergey-shandar — APPROVED** [review]

File: stacks-coordinator/src/bitcoin_node.rs [hunk 1]
```diff
@@ -169,6 +169,7 @@ impl LocalhostBitcoinNode {
             .map(|path| Cow::Owned(format!("{}{}", &self.bitcoind_api, path)))
```
Comment (netrome) [inline stacks-coordinator/src/bitcoin_node.rs:169]: Nit: Just pasting @stjepangolemac's suggestion here to make it easier to commit.
```suggestion
            .map(|path| {
                Cow::Owned(format!(
                    "{}/{}",
                    &self.bitcoind_api,
                    path.trim_start_matches('/')
                ))
            })
            .unwrap_or(Cow::Borrowed(&self.bitcoind_api));
```

File: stacks-coordinator/src/bitcoin_node.rs [hunk 2]
```diff
@@ -150,7 +150,7 @@ impl LocalhostBitcoinNode {
         self.call_path(
             method,
             params,
-            Some(&format!("wallet/{}", self.wallet_name)),
+            Some(&format!("/wallet/{}", self.wallet_name)),
```
Comment (netrome) [inline stacks-coordinator/src/bitcoin_node.rs:153]: Also, no need to change this if incorporating the update below
```suggestion
            Some(&format!("wallet/{}", self.wallet_name)),
```

**Review by netrome — APPROVED** [review]

**Comment by jferrant** [comment]
> Sorry we fixed this today but forgot to push 🤦
> 
> This might make the `url` more resistant to misuse:
> 
> ```
> let url = path
>             .map(|path| {
>                 Cow::Owned(format!(
>                     "{}/{}",
>                     &self.bitcoind_api,
>                     path.trim_start_matches('/')
>                 ))
>             })
>             .unwrap_or(Cow::Borrowed(&self.bitcoind_api));
> ```

Haha no worries. I ended up doing a more robust solution that utilizes URL so we don't have to worry as much about missing or extra "/" throughout the code.

File: stacks-coordinator/src/bitcoin_node.rs [hunk 3]
```diff
@@ -165,11 +168,14 @@ impl LocalhostBitcoinNode {
...
-            .map(|path| Cow::Owned(format!("{}{}", &self.bitcoind_api, path)))
-            .unwrap_or(Cow::Borrowed(&self.bitcoind_api));
+        let url = if let Some(path) = path {
+            Cow::Owned(self.bitcoind_api.join(path)?)
+        } else {
+            Cow::Borrowed(&self.bitcoind_api)
+        };
 
-        let response = ureq::post(&url)
+        println!("Calling {} ...", url);
```
Comment (netrome) [inline stacks-coordinator/src/bitcoin_node.rs:177]: Nit: Remove or change to debug logging
```suggestion
        tracing::debug!("Calling url: {}.", url);
```

**Review by netrome — APPROVED** [review]

File: stacks-coordinator/src/bitcoin_node.rs (hunk 3 above)
Comment (jferrant) [reply stacks-coordinator/src/bitcoin_node.rs]: Good catch :)

**Pull request merged**